src/core/ext/filters/channel_idle/channel_idle_filter.cc

```diff
@@ -277,9 +278,10 @@ void ChannelIdleFilter::StartIdleTimer() {
 
 void ChannelIdleFilter::CloseChannel() {
   auto* op = grpc_make_transport_op(nullptr);
-  op->disconnect_with_error = grpc_error_set_int(
-      GRPC_ERROR_CREATE("enter idle"),
-      StatusIntProperty::ChannelConnectivityState, GRPC_CHANNEL_IDLE);
+  op->disconnect_with_error =
+      GRPC_ERROR_BUILDER(kUnknown, "enter idle")
```

Comment:
Suggest `ErrorBuilder()`

Reply:
If you suggest to not use a macro, it's not feasible without `absl::SourceLocation` which can fetch this information without relying on macro.

Reply:
That's spelled `grpc_core::SourceLocation` and we have it.

Reply:
Oh that's amazing. But this doesn't seem to work on all platforms gRPC supports. Two things to do;
 1. Extending the check to use __builtin_LINE. (e.g. gcc 6 has it but it doesn't has `__has_builtin` function so we can allow it if gcc >= 6 even when there is no `__has_builtin`)
 2. Having a consensus on whether this source_location information can be omitted in certain environment such as msvc 2017.

In case of grpc_error, I guess error location isn't that critical so I'm fine with optional absence of it.